recipes/libigl/2.x.x/conanfile.py

```diff
@@ -42,130 +32,110 @@ def _minimum_cpp_standard(self):
     def _minimum_compilers_version(self):
         return {
             "Visual Studio": "16",
+            "msvc": "1920",
             "gcc": "6",
             "clang": "3.4",
             "apple-clang": "5.1",
         }
 
-    def validate(self):
-        if self.settings.compiler.get_safe("cppstd"):
-            check_min_cppstd(self, self._minimum_cpp_standard)
-        min_version = self._minimum_compilers_version.get(str(self.settings.compiler))
-        if not min_version:
-            self.output.warn(
-                "{} recipe lacks information about the {} compiler support.".format(
-                    self.name, self.settings.compiler
-                )
-            )
-        else:
-            if Version(self.settings.compiler.version) < min_version:
-                raise ConanInvalidConfiguration(
-                    "{} requires C++{} support. The current compiler {} {} does not support it.".format(
-                        self.name,
-                        self._minimum_cpp_standard,
-                        self.settings.compiler,
-                        self.settings.compiler.version,
-                    )
-                )
-        if (
-            self.settings.compiler == "Visual Studio"
-            and "MT" in self.settings.compiler.runtime
-            and not self.options.header_only
-        ):
-            raise ConanInvalidConfiguration(
-                "Visual Studio build with MT runtime is not supported"
-            )
-        if ("arm" in self.settings.arch or "x86" is self.settings.arch) and (
-            Version(self.version) < "2.4.0" or not is_apple_os(self)
-        ):
-            raise ConanInvalidConfiguration(
-                "Not available for arm. Requested arch: {}, os: {}".format(
-                    self.settings.arch, self.settings.os
-                )
-            )
+    def export_sources(self):
+        export_conandata_patches(self)
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
     def configure(self):
         if self.options.header_only:
-            del self.options.fPIC
+            try:
+                del self.options.fPIC
+            except Exception:
+                pass
+
+    def layout(self):
+        if not self.options.header_only:
+            cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        self.requires("eigen/3.4.0")
```

Comment:
On the libigl 2.4.0 branch, the eigen pulled in is 3.3.7 (through FetchContent). Does it make sense to provide a different version here? (it also has some breaking changes)

Reply:
Let's try 3.4.0 Using old versions we risking a package version conflict, which will break any build.